operator/option/config.go

```diff
@@ -450,6 +453,11 @@ type OperatorConfig struct {
 	// The default is zero if this option is not specified.
 	IngressProxyXffNumTrustedHops uint32
 
+	// GatewayAPIProxyXffNumTrustedHops The number of additional GatewayAPI proxy hops from the right side of the
+	// HTTP header to trust when determining the origin client's IP address.
+	// The default is zero if this option is not specified.
+	GatewayAPIProxyXffNumTrustedHops uint32
+
```

Comment:
Shouldn't this be declared as a GatewayAPI "private" config option in `operator/pkg/gateway-api/cell.go` (e.g: like `GatewayAPISecretsNamespace`)? I don't see any other usage outside of the Gateway API cell.

Not related to this PR, but I think it's the same for `IngressDefaultXffNumTrustedHops`.

Reply:
Done.

Reply:
>. Not related to this PR, but I think it's the same for IngressDefaultXffNumTrustedHops.

I will make this change in the next PR.

Reply:
> I will make this change in the next PR.

Thanks! :pray: